deno_dist/helper/ssg/index.ts

```diff
@@ -25,57 +25,93 @@ export interface ToSSGResult {
   error?: Error
 }
 
-const generateFilePath = (routePath: string, outDir: string) => {
-  const fileName = routePath === '/' ? 'index.html' : routePath + '.html'
+const generateFilePath = (routePath: string, outDir: string, mimeType: string) => {
+  const extension = determineExtension(mimeType)
+  const fileName = routePath === '/' ? `index${extension}` : `${routePath}${extension}`
   return joinPaths(outDir, fileName)
 }
 
+const parseResponseContent = async (response: Response): Promise<string | ArrayBuffer> => {
+  const contentType = response.headers.get('Content-Type')
+
+  try {
+    if (contentType?.includes('text')) {
+      return await response.text()
+    } else if (contentType?.includes('json')) {
```

Comment:
Maybe we can use `response.text()` for json also? I don't think it is necessary to decode JSON and then turn it into a string.